src/openms/source/ANALYSIS/ID/FalseDiscoveryRate.cpp

```diff
@@ -485,17 +486,17 @@ namespace OpenMS
       it->setHigherScoreBetter(false);
       const vector<ProteinHit>& old_hits = it->getHits();
       vector<ProteinHit> new_hits;
-      for (auto hit : old_hits)
+      for (auto hit : old_hits) // NOTE: performs copy
       {
         // Add decoy proteins only if add_decoy_proteins is set
         if (add_decoy_proteins || hit.getMetaValue("target_decoy") != "decoy")
         {
           hit.setMetaValue(score_type, hit.getScore());
           hit.setScore(score_to_fdr[hit.getScore()]);
-          new_hits.push_back(hit);
+          new_hits.push_back(std::move(hit));
         }
       }
-      it->setHits(new_hits);
+      it->setHits(std::move(new_hits));
```

Comment:
I agree, it does not help right now and doesnt hurt, but it will be better once we have an implementation of `void setHits(std::vector<ProteinHit>&& hits);`. Also I think it makes the meaning clear but I agree your solution is better.